tests/unit_tests/test_source_mesh.py

```diff
@@ -0,0 +1,148 @@
+    # skip the test if the library is not enabled
+    if test_cases['library'] == 'moab' and not openmc.lib._dagmc_enabled():
+        pytest.skip("DAGMC (and MOAB) mesh not enabled in this build.")
+
+    if test_cases['library'] == 'libmesh' and not openmc.lib._libmesh_enabled():
+        pytest.skip("LibMesh is not enabled in this build.")
+
+    # setup mesh source ###
+    mesh_filename = Path(request.fspath).parent / "test_mesh_tets.e"
+    uscd_mesh = openmc.UnstructuredMesh(mesh_filename, test_cases['library'])
+
+    # subtract one to account for root cell produced by RegularMesh.build_cells
+    n_cells = len(model.geometry.get_all_cells()) - 1
+
+    # set source weights according to test case
+    if test_cases['source_strengths'] == 'uniform':
+        vol_norm = True
+        strengths = None
+    elif test_cases['source_strengths'] == 'manual':
+        vol_norm = False
+        # assign random weights
+        strengths = np.random.rand(n_cells*TETS_PER_VOXEL)
+
+    # create the spatial distribution based on the mesh
+    space = openmc.stats.MeshSpatial(uscd_mesh, strengths, vol_norm)
+
+    energy = openmc.stats.Discrete(x=[15.e+06], p=[1.0])
+    source = openmc.Source(space=space, energy=energy)
+    model.settings.source = source
+
+    with cdtemp([mesh_filename]):
+        model.export_to_xml()
+
+        n_measurements = 100
+        n_samples = 1000
+
+        cell_counts = np.zeros((n_cells, n_measurements))
+
+        # This model contains 1000 geometry cells. Each cell is a hex
+        # corresponding to 12 of the tets. This test runs 10000 particles. This
+        #  results in the following average for each cell
+        average_in_hex = n_samples / n_cells
+
+        openmc.lib.init([])
+
+        # perform many sets of samples and track counts for each cell
+        for m in range(n_measurements):
+            sites = openmc.lib.sample_external_source(n_samples)
+            cells = [openmc.lib.find_cell(s.r) for s in sites]
+
+            for c in cells:
+                # subtract one from index to account for root cell
+                cell_counts[c[0]._index - 1, m] += 1
+
+        openmc.lib.finalize()
+
+        # normalize cell counts to get sampling frequency per particle
+        cell_counts /= n_samples
+
+        # get the mean and std. dev. of the cell counts
+        mean = cell_counts.mean(axis=1)
+        std_dev = cell_counts.std(axis=1)
+
+        if test_cases['source_strengths'] == 'uniform':
+            exp_vals = np.ones(n_cells) / n_cells
+        else:
+            # sum up the source strengths for each tet, these are the expected true mean
+            # of the sampling frequency for that cell
+            exp_vals = strengths.reshape(-1, 12).sum(axis=1) / sum(strengths)
+
+        diff = np.abs(mean - exp_vals)
+        assert((diff < 2*std_dev).sum() / diff[:10].size >= 0.95)
+        assert((diff < 6*std_dev).sum() / diff.size >= 0.97)
```

Comment:
Neither of these seem to make sense to me -- can you explain?

Reply:
We talked about these offline, but for posterity:
   - The [:10] was leftover from debugging an issue. My bad
   - The `0.97` was a typo and should be `0.997`.